ts/nni_manager/training_service/v3/compat.ts

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { EventEmitter } from 'events';
+import { readFile } from 'fs/promises';
+import path from 'path';
+
+import { Deferred } from 'common/deferred';
+import type { TrainingServiceConfig } from 'common/experimentConfig';
+import globals from 'common/globals';
+import {
+    TrainingService, TrialJobApplicationForm, TrialJobDetail, TrialJobMetric, TrialJobStatus
+} from 'common/trainingService';
+import type { EnvironmentInfo, Parameter, TrainingServiceV3 } from 'common/training_service_v3';
+import { trainingServiceFactoryV3 } from './factory';
+
+type MutableTrialJobDetail = {
+    -readonly [Property in keyof TrialJobDetail]: TrialJobDetail[Property];
+};
+
+export class V3asV1 implements TrainingService {
```

Comment:
What is `V3asV1` used for? I thought it should be scheduler who interacts with TrainingServiceV3. But here, it seems to be V3asV1 who interacts with TrainingServiceV3.

Reply:
The compatibility layer is required to run local v3 integration tests before the new scheduler ready.

Reply:
got it